### PR TITLE
DS-1201: Workaround for DELETE resources with BatchRequests issue

### PR DESCRIPTION
**Pull request by anupama-pathirage**

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBDeployer.java [hunk 1]
```diff
@@ -849,7 +849,8 @@ public int compare(Pattern o1, Pattern o2) {
 
 			/* REST processing - adding DS resources to AxisService */
 			Set<ResourceID> resourceIds = dataService.getResourceIds();
-			for (ResourceID resourceId : resourceIds) {
+            Set<ResourceID> sortedResourceIds = new TreeSet<>(resourceIds).descendingSet();
```
Comment (chanikag) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBDeployer.java]: Code formatting - indent

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/resource/Resource.java [hunk 2]
```diff
@@ -104,7 +104,13 @@ public boolean equals(Object rhs) {
...
-	}
-	
+
+        @Override
+        public int compareTo(Object obj) {
+            ResourceID resource = (ResourceID) obj;
+            return generateRequestName(this.path, this.method).
+                    compareTo(generateRequestName(resource.getPath(), resource.getMethod()));
+        }
+
```
Comment (swsachith) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/resource/Resource.java]: Unnecessary spaces.

Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/description/resource/Resource.java]: Fixed with PR: #129

File: components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBDeployer.java (hunk 1 above)
Comment (anupama-pathirage) [inline components/data-services/org.wso2.carbon.dataservices.core/src/main/java/org/wso2/carbon/dataservices/core/DBDeployer.java]: Fixed with PR: #129

**Pull request merged**